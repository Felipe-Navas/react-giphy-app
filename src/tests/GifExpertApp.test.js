import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';

import { GifExpertApp } from '../GifExpertApp';

describe('Testing the <GifExpertApp /> component', () => {
  test('should render', () => {
    const wrapper = shallow(<GifExpertApp defaultCategories={['One Punch']} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('should display a list of categories', () => {
    const categories = ['Fiat', 'Chebrolet'];
    const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('GifGrid').length).toBe(categories.length);
  });
});
