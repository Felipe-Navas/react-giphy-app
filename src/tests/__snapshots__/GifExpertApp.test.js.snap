// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the <GifExpertApp /> component should display a list of categories 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Fiat"
      key="Fiat"
    />
    <GifGrid
      category="Chebrolet"
      key="Chebrolet"
    />
  </ol>
</Fragment>
`;

exports[`Testing the <GifExpertApp /> component should render 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
  </ol>
</Fragment>
`;
